tokio-util/src/net/mod.rs

```diff
@@ -39,9 +39,16 @@ impl Listener for tokio::net::TcpListener {
         Self::poll_accept(self, cx)
     }
 
+    #[cfg(not(target_os = "wasi"))]
     fn local_addr(&self) -> Result<Self::Addr> {
         self.local_addr().map(Into::into)
     }
+
+    #[cfg(target_os = "wasi")]
+    fn local_addr(&self) -> Result<Self::Addr> {
+        use std::io;
+        Err(io::ErrorKind::Unsupported.into())
+    }
```

Comment:
Please put the cfg inside the function.
```suggestion
    fn local_addr(&self) -> Result<Self::Addr> {
        #[cfg(not(target_os = "wasi"))]
        self.local_addr().map(Into::into)

        #[cfg(target_os = "wasi")]
        Err(std::io::ErrorKind::Unsupported.into())
    }
```